### PR TITLE
Update grammar to 2.6

### PR DESCRIPTION
**Pull request by iliabylich**
Closes https://github.com/sorbet/sorbet/issues/972

### Motivation
High-level changes:
1. Bump `TARGET_RUBY_VERSION` to `2.6` in `tools/scripts/import_whitequark.sh`
2. Re-generated tests in `test/whitequark`
3. Changed `typedruby.ypp` to match 2.6
4. Updated a few existing tests with the code that is invalid for 2.6
6. Slightly updated exclude filters in `tools/scripts/import_whitequark.rb`

Changes to grammar:
1. useless `begin; else; end` is now a syntax error (previously it was a warning)
2. `-> do rescue; end` is supported now
3. endless ranges (`(1..)`) are supported now. I didn't have to change anything except the grammar because ranges are generic (`Range<T>.begin` returns `T`), and so `(1..).begin` returns `Integer | NilClass`. Later it can be extended (maybe by having multiple subclasses or by overloading a constructor like `takes T, T -> [begin = T, end = T]` / `takes T, nil -> [begin = T, end = NilClass]`)

### Test plan

See included automated tests.

### CODE REVIEW

File: test/testdata/deviations/keyword_method_names.rb [hunk 1]
```diff
@@ -73,14 +73,6 @@ def method_named_do(x)
     do # error: unexpected token
 end
 
-def method_named_else(x)
```
Comment (DarkDimius) [inline test/testdata/deviations/keyword_method_names.rb]: why was this deleted?

Comment (iliabylich) [reply test/testdata/deviations/keyword_method_names.rb]: > 1. `useless begin; else; end` is now a syntax error (previously it was a warning)

because it produces a syntax error that is completely unrelated to this test

Comment (DarkDimius) [reply test/testdata/deviations/keyword_method_names.rb]: does it? It checks that we can call a method called `else`.
Can we keep it?

Comment (iliabylich) [reply test/testdata/deviations/keyword_method_names.rb]: I'm not 100% sure, but I thought this test asserts that the code like
```
a.
  <keyword>
```
is interpreted as `a.<keyword>` (and not as `a.<keyword as a method name>`) to speedup language server.

It works with `else`, but `begin; else; end` is a syntax error now, and so
```
a.
  else
```
becomes `tIDENTIFIER(A) -> tDOT -> tELSE` which gives a different error now.

I can update an error message, but I don't know if the language server still works for this case. @jez Could you take a look please?

Comment (iliabylich) [reply test/testdata/deviations/keyword_method_names.rb]: @DarkDimius I've just added back this test. Here's the original method:
``` ruby
def method_named_else(x)
  x.else # ok
  x
    .else # ok
  x.
    else # error: unexpected token
end
```

The last part now produces an error, and so the whole method body gets reduced into an `error` (and it's completely useless now). I added back first two statements. Also, the CI seems to failing now. I didn't change anything except input/output files, but it fails in the completely unrelated job.

Comment (DarkDimius) [reply test/testdata/deviations/keyword_method_names.rb]: >  I added back first two statements. Also, the CI seems to failing now.

you'll need to rebase, we did some changes to the compilation toolchain for the weekend that aren't correctly invalidated by the build cache.

Comment (iliabylich) [reply test/testdata/deviations/keyword_method_names.rb]: @DarkDimius done

**Comment by ptarjan** [comment]
Master was broken (emscripten moved to GCP from AWS). If you rebase, this looks great to merge. Thanks so much for the contribution!
